src/migration/1727578711381-VordrUser.ts

```diff
@@ -0,0 +1,90 @@
+import { MigrationInterface, QueryRunner } from "typeorm";
+
+export class VordrUser1727578711381 implements MigrationInterface {
+  name = 'VordrUser1727578711381'
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(`CREATE INDEX "IDX_post_flags_vordr" ON post (((flags->'vordr')::boolean))`);
+
+    await queryRunner.query(`CREATE OR REPLACE FUNCTION vordr_update_flags()
+RETURNS TRIGGER AS $$
+BEGIN
+  IF NEW.flags @> '{"vordr": true}' THEN
+    -- If vordr on user is set to true
+    -- Set banned and vordr to true in post table, and set vordr to true in comment table
+    UPDATE
+      post
+    SET
+      "banned" = TRUE,
+      "flags" = jsonb_set(jsonb_set(flags,
+        '{vordr}', 'true', true),         -- set vordr to true
+        '{banned}', 'true', true)         -- set banned to true
+    WHERE
+      ("authorId" = NEW.id OR "scoutId" = NEW.id);
+
+    UPDATE
+      comment
+    SET
+      "flags" = jsonb_set(flags, '{vordr}', 'true', true)
+    WHERE
+      "userId" = NEW.id;
+
+  -- If vordr on user is set to false, and vordr on post is true
+  -- Set banned and vordr to false in post table, and set vordr to false in comment table
+  ELSIF NEW.flags @> '{"vordr": false}' THEN
+    UPDATE
+      post
+    SET
+      "banned" = FALSE,
+      "flags" = jsonb_set(jsonb_set(flags,
+        '{vordr}', 'false', true),        -- set vordr to false
+        '{banned}', 'false', true)        -- set banned to false
+    WHERE
+      ("authorId" = NEW.id OR "scoutId" = NEW.id)
+      AND (flags->>'vordr')::boolean = true;
+
+    UPDATE
+      comment
+    SET
+      "flags" = jsonb_set(flags, '{vordr}', 'false', true)
+    WHERE
+      "userId" = NEW.id;
+  END IF;
+  RETURN NEW;
+END;
+$$ LANGUAGE plpgsql;`);
+
+    await queryRunner.query(`CREATE OR REPLACE TRIGGER user_flags_vordr
+AFTER UPDATE OF flags ON "user"
+FOR EACH ROW
+WHEN ((OLD.flags ->> 'vordr')::boolean IS DISTINCT FROM ((NEW.flags ->> 'vordr'))::boolean)
```

Comment:
You don't have to cast to string `->>` and then `boolean` again? Using only `((OLD.flags -> 'vordr')::boolean` should be fine?

Reply:
Also `IS DISTINCT` works when flags->vordr is null/empty?

Reply:
> Also IS DISTINCT works when flags->vordr is null/empty?

Pretty sure, could add test case for it

Reply:
> You don't have to cast to string ->> and then boolean again? Using only ((OLD.flags -> 'vordr')::boolean should be fine?

Changed it to just be `OLD.flags -> 'vordr' IS DISTINCT FROM NEW.flags -> 'vordr'`

Reply:
> Also IS DISTINCT works when flags->vordr is null/empty?
Yes.

It is already in the tests when changing userId 1, it doesn't have vordr flag set. Or do you want explicit test for it?

Reply:
That's fine if works 👍 Was more concerned for this case https://github.com/dailydotdev/daily-api/pull/2289#discussion_r1784296841